python/ray/_private/metrics.py

```diff
@@ -0,0 +1,67 @@
+import functools
+import time
+from ray.util.metrics import Counter, Histogram
+from typing import Dict, Callable
+
+
+class FunctionMetricsCollector:
+    """A helper class that implements the context manager and decorator
+    for recording function/block execution.
+    """
+    HISTOGRAMS: Dict[str, Histogram] = dict()
+    SUCCESS_COUNTERS: Dict[str, Counter] = dict()
+    FAILURE_COUNTERS: Dict[str, Counter] = dict()
+
+    def __init__(self, name: str):
+        self._latency_historgram = self.get_latency_histogram(name)
+        self._success_counter = self.get_success_counter(name)
+        self._failure_counter = self.get_failure_counter(name)
+
+    def __enter__(self):
+        self._start_time = time.time()
```

Comment:
Maybe using `time.perf_counter`? Afaik, it has better resolution https://docs.python.org/3/library/time.html#time.perf_counter

Reply:
The general recommendation is to use `timeit.default_timer()`: https://docs.python.org/3/library/timeit.html#timeit.default_timer